hudi-common/src/main/java/org/apache/hudi/common/table/view/HoodieTableFileSystemView.java

```diff
@@ -398,14 +398,24 @@ protected Option<Pair<String, CompactionOperation>> getPendingLogCompactionOpera
 
   @Override
   protected boolean isPartitionAvailableInStore(String partitionPath) {
-    return partitionToFileGroupsMap.containsKey(partitionPath);
+    try {
+      readLock.lock();
+      return partitionToFileGroupsMap.containsKey(partitionPath);
```

Comment:
This method is internal, the invoker should hold the lock.

Reply:
I don't think this is true, why does it matter if it is internal? The call to this method can happen outside of the read lock and it is requiring a read from the map

Reply:
because public APIs are designed for interpalying with the class and the for internal methods, the entrance methods(as public) should take care of the acquisition of the read/write lock.

Reply:
I don't think this is actually accurate. If you move the locks closer to the places that the access happens you can improve the throughput for the class. By saying you can only use locks in public methods, you really limit yourself.